server/internal/infrastructure/memory/model_test.go

```diff
@@ -0,0 +1,146 @@
+package memory
+
+import (
+	"context"
+	"testing"
+
+	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/model"
+	"github.com/reearth/reearthx/rerror"
+	"github.com/reearth/reearthx/util"
+	"github.com/reearth/reearthx/usecasex"
+
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewModel(t *testing.T) {
+	expected := &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		now:  &util.TimeNow{},
+	}
+	got := NewModel()
+	assert.Equal(t, expected, got)
+}
+
+func TestMemory_Filtered(t *testing.T) {
+	r := &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+	}
+	expected := &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    r.f.Merge(r.f),
+		now:  &util.TimeNow{},
+	}
+	got := r.Filtered(r.f)
+	assert.Equal(t, expected, got)
+}
+
+func TestMemory_FindByProject(t *testing.T) {
+	ctx := context.Background()
+	pId := id.NewProjectID()
+	r := &Model{
+		data: &util.SyncMap[id.ModelID, *model.Model]{},
+		f:    repo.WorkspaceFilter{},
+	}
+	expectedModelList := model.List(r.data.FindAll(func(_ id.ModelID, m *model.Model) bool {
+		return m.Project().Equal(pId)
+	})).SortByID()
+	var startCursor, endCursor *usecasex.Cursor
+	expectedPageInfo := usecasex.NewPageInfo(
+		0,
+		startCursor,
+		endCursor,
+		true,
+		true,
+	)
+	var arg *usecasex.Cursor
+	gotModelList, gotPageInfo, err := r.FindByProject(ctx, pId, usecasex.NewPagination(lo.ToPtr(1), lo.ToPtr(1), arg, arg))
+	assert.NoError(t, err)
+	assert.Equal(t, expectedModelList, gotModelList)
+	assert.Equal(t, expectedPageInfo, gotPageInfo)
+}
```

Comment:
To cover all the source code you might seed the data with some project. some off code will be executed only if there is returned data.